src/test/groovy/com/commercehub/gradle/plugin/avro/OptionsFunctionalSpec.groovy

```diff
@@ -299,4 +305,50 @@ class OptionsFunctionalSpec extends FunctionalSpec {
         then:
         taskInfoAbsent || result.task(":generateAvroJava").outcome == SUCCESS
     }
+
+    @Unroll
+    def "supports configuration of dateTimeLogicalType to #dateTimeLogicalType"() {
+        given:
+        copyResource("user.avsc", avroDir)
+        buildFile << """
+        |avro {
+        |    dateTimeLogicalType = "${dateTimeLogicalType}"
+        |}
+        |""".stripMargin()
+
+        when:
+        def result = run("generateAvroJava")
+
+        then: "the task succeeds"
+        taskInfoAbsent || result.task(":generateAvroJava").outcome == SUCCESS
+        def content = projectFile("build/generated-main-avro-java/example/avro/User.java").text
+
+        and: "the specified dateTimeLogicalType is used"
+        content.contains("public void setBirthDate(${fieldClz} value)")
+
+        where:
+        dateTimeLogicalType                                                              | fieldClz
+        SpecificCompiler.DateTimeLogicalTypeImplementation.JODA.name()                   | "org.joda.time.LocalDate"
+        SpecificCompiler.DateTimeLogicalTypeImplementation.JODA.name().toLowerCase()     | "org.joda.time.LocalDate"
+        SpecificCompiler.DateTimeLogicalTypeImplementation.JSR310.name()                 | "java.time.LocalDate"
+        SpecificCompiler.DateTimeLogicalTypeImplementation.DEFAULT.name()                | actualDateTimeImplementationDefault
+        SpecificCompiler.DateTimeLogicalTypeImplementation.DEFAULT.name().toLowerCase()  | actualDateTimeImplementationDefault
+    }
+
+    def "rejects unsupported dateTimeLogicalType values"() {
+        given:
+        copyResource("user.avsc", avroDir)
+        buildFile << """
+        |avro {
+        |    dateTimeLogicalType = "badValue"
+        |}
+        |""".stripMargin()
+
+        when:
+        def result = runAndFail("generateAvroJava")
+
+        then:
+        taskInfoAbsent || result.task(":generateAvroJava").outcome == FAILED
+        result.output.contains("Invalid dateTimeLogicalType 'badValue'.  Value values are: [JODA, JSR310]")
```

Comment:
Because this message is generated out of the `Enum.parseCaseInsensitive` method, it does not include DEFAULT, which unfortunately Avro upstream did not include as part of the Enum itself. Either we can remove DEFAULT as a valid value for this configuration setting, OR we have to special case this error message, OR we live with the error message missing the `DEFAULT` value as a valid value. Let me know what you want to do here.

Reply:
Let's not include "DEFAULT" as a valid value.  If a user wants the default behavior, they simply don't configure anything.

Reply:
Done